src/find/matchers/mod.rs

```diff
@@ -361,15 +369,24 @@ fn build_matcher_tree(
     let mut invert_next_matcher = false;
     while i < args.len() {
         let possible_submatcher = match args[i] {
-            "-print" => Some(Printer::new(PrintDelimiter::Newline).into_box()),
-            "-print0" => Some(Printer::new(PrintDelimiter::Null).into_box()),
+            "-print" => Some(Printer::new(PrintDelimiter::Newline, None).into_box()),
+            "-print0" => Some(Printer::new(PrintDelimiter::Null, None).into_box()),
             "-printf" => {
                 if i >= args.len() - 1 {
                     return Err(From::from(format!("missing argument to {}", args[i])));
                 }
                 i += 1;
                 Some(Printf::new(args[i])?.into_box())
             }
+            "-fprint" => {
+                if i >= args.len() - 1 {
+                    return Err(From::from(format!("missing argument to {}", args[i])));
+                }
+                i += 1;
+
+                let file = get_or_create_file(args[i])?;
```

Comment:
For future reference: GNU find de-duplicates files, so if you do `find -fprint foo -fprint foo` it will use the same `FILE *` for both and you'll get each line twice.  Right now uutils will use two different `File`s and the writes will overlap with each other.